docker-compose.yml

```diff
@@ -0,0 +1,41 @@
+version: '3.8'
+
+services:
+  mysql:
+    image: mysql:9.0.0
+    container_name: mysql-cont
+    ports:
+      - "3307:3307"
+    environment:
+      MYSQL_ROOT_PASSWORD: root
+      MYSQL_DATABASE: usersystem
+      MYSQL_PASSWORD: root
+      MYSQL_TCP_PORT: 3307
+    volumes:
+      - mysql-data:/var/lib/mysql
+    networks:
+      - app-network
+    restart: always  # Restart policy for MySQL
+
+  spring-boot-app:
+    build: .
+    container_name: spring-boot-app
+    ports:
+      - "8080:8080"
+    environment:
+      SPRING_DATASOURCE_URL: jdbc:mysql://mysql:3307/usersystem?useSSL=false&serverTimezone=UTC&createDatabaseIfNotExist=true&allowPublicKeyRetrieval=true
+      SPRING_DATASOURCE_USERNAME: root
+      SPRING_DATASOURCE_PASSWORD: root
```

Comment:
Its very important not to expose any password in code changes pushed to GitHub.
Using the .env file can help solve these issues.